### DISCUSSION TITLE
Computation of viscous drag load from hybrid modelling in HydroDyn.

### DISCUSSION THREAD

**Discussion by darkstar71** [OpenFAST/openfast#2364] [Q&A] [answered]
Hi,
I was trying to estimate RAOs in surge, heave and pitch of a semi-submersible concept (similar to the DeepC platform) for wave-only load cases using OpenFAST. I am interested to know

1. how to activate/deactivate the computation of viscous drag while running a coupled simulation?
2. how to obtain a time history of viscous drag force acting on the floating platform (not on a single member) in the vertical direction?

**Comment by jjonkman** [comment]
Dear @darkstar71,

Here are my responses:

1. Hydrodynamic viscous drag is implemented in HydroDyn either through the strip-theory solution (by specifying drag coefficients (Cd) for each member (transverse, axial), and joint (axial) and/or through the additional quadratic drag matrix (`AddBQuad`) augmented to the potential-flow solution, with the former method generally preferred.
2. HydroDyn does not have a direct output for the total viscous drag in the vertical direction.  But you could likely infer it from other outputs, e.g., if viscous drag is the only hydrodynamic load not accounted for in the potential-flow solution, you could subtract the potential-flow contributions from `HydroFzi` to arrive at it.

Best regards,

**Reply by darkstar71** [reply]
Dear @jjonkman,
Thank you very much for your answer.

Regarding your reply on question 2, my understanding for calculation of vertical hydrodynamic viscous drag force is as follows:

potential-flow contributions = WavesF1zi + RdtnFzi 
viscous-drag contribution = HydroFzi - (WavesF1zi + RdtnFzi)

Please comment.

best regards,

**Reply by jjonkman** [reply, marked as answer]
Dear @darkstar71,

You have the right idea, but the potential-flow output names you are referring to from HydroDyn are outdated; also, the potential-flow solution also considers hydrostatics and potentially "additional loads".  Assuming that you are using the newest version of OpenFAST, that your HydroDyn model has only one potential-flow body, and that viscous drag is the only hydrodynamic load not accounted for in the potential-flow solution, the math would be:

viscous contribution in zi direction = `HydroFzi` - ( `B1AddFzi` + `B1WvsFzi` + `B1HdSFzi` + `B1RdtFzi` )

Best regards,
